objcgen/objcgenerator.cs

```diff
@@ -522,7 +530,7 @@ string GenerateArgumentArrayPost (string parameterName, string argumentName, Typ
 				break;
 			case TypeCode.DateTime:
 				postwriter.WriteLine ($"E4KDateTime {presarrval} = mono_array_get ({presarr}, E4KDateTime, {pindex});");
-				postwriter.WriteLine ($"{presobj} = mono_embeddinator_get_nsdate (&{presarrval});");
+				postwriter.WriteLine ($"{presobj} = mono_embeddinator_get_nsdate ((E4KDateTime *) &{presarrval});");
```

Comment:
Why is this needed if the variable is declared with the right type on the previous line?

I assume you got a compiler error, but I can't see from the change why it would fail to compile.

Reply:
I must have seen other mono_embeddinator_get_nsdate bits in the patch and added an unnecessary cast here. Checking others.